File: WebApplication/src/approve-registrations/filter-dtos/approve-registration-filter.dto.ts
import { FilterDto } from "src/shared/filter-dtos/filter.dto";
import { ApproveRegistrationState } from "../enums/approve-registration-state.enum";
import { InstitutionDto } from "src/nomenclatures/dtos/institutions/institution.dto";
import { ComplexDto } from "src/nomenclatures/dtos/complexes/complex.dto";

export class ApproveRegistrationFilterDto extends FilterDto {
  authorizedRepresentativeUsername: string;
  authorizedRepresentativeFullname: string;
  institutionId: number;
  institution: InstitutionDto;
  complexId: number;
  complex: ComplexDto;
  administratedUserId: number;
  state: ApproveRegistrationState;
}

File: Sc.Models/FilterDtos/ApproveRegistrations/ApproveRegistrationFilterDto.cs
using Sc.Models.Entities.ApproveRegistrations;
using Sc.Models.Enums.State;
using Sc.Models.Filters.Base;

namespace Sc.Models.FilterDtos.ApproveRegistrations
{
    public class ApproveRegistrationFilterDto : FilterDto<ApproveRegistration>
    {
        public string AuthorizedRepresentativeUsername { get; set; }
        public string AuthorizedRepresentativeFullname { get; set; }
        public int? InstitutionId { get; set; }
        public int? ComplexId { get; set; }
        public int? AdministratedUserId { get; set; }
        public ApproveRegistrationState? State { get; set; }

        public override IQueryable<ApproveRegistration> WhereBuilder(IQueryable<ApproveRegistration> query)
        {
            if (AdministratedUserId.HasValue)
            {
                query = query.Where(e => e.AdministratedUserId == AdministratedUserId);
            }

            if (State.HasValue)
            {
                query = query.Where(e => e.State == State);
            }

            return base.WhereBuilder(query);
        }
    }
}
